crates/bevy_ecs/src/system/input.rs

```diff
@@ -180,3 +180,25 @@ impl<E: 'static, B: Bundle> SystemInput for Trigger<'_, E, B> {
         this
     }
 }
+
+/// A helper for using [`SystemInput`]s in generic contexts.
+///
+/// This type is a [`SystemInput`] adapter which always has
+/// `Self::Param::Inner == Self::Inner`.
```

Comment:
I think the crucial point is that `Self::Param == Self`, since that's effectively the requirement for implementing `SystemParamFunction`. Could you also add a mention of the fact that this ignoring the lifetimes?

I checked what the docs for `StaticSystemParam` say and they mention `Self::State::Item == Self`, which seem wrong to me, it should be `Self::Item == Self`. That should also need to be fixed.